src/main/java/Duke.java

```diff
@@ -1,10 +1,122 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static String line = "__________________________________________________________";
+    private static Task[] inputList = new Task[101];
+    private static int numTasks = 0;
+
+    public static void printAddTask(Task t) {
+        //print to show added to list
+        System.out.println(line);
+        System.out.println("Got it. I've added this task: \n" + t);
+        System.out.println("Now you have " + numTasks + " tasks in the list.");
+        System.out.println(line);
+    }
+
+    public static void validTask(String[] userInput) throws DukeException{
+        if (userInput.length < 2 && (userInput[0].equals("todo") ||
+                userInput[0].equals("event") || userInput[0].equals("deadline"))) {
+            throw new DukeException();
+        }
+    }
+
+    public static void validTask(String userInput) throws DukeException {
+        String taskNum = userInput.substring(userInput.length()-1);
+        int x = Integer.parseInt(taskNum);
+        if (inputList[x] == null) {
+            throw new DukeException();
+        }
+        inputList[x].markAsDone(userInput);
+    }
+
+    public static void addTask(String userInput) throws DukeException {
+        Task t;
+        String[] words = userInput.split(" ");
+        validTask(words);
+        String descriptor = userInput.substring(userInput.indexOf(words[1]), userInput.length());
+        if (words[0].equals("todo")) {
+            t = new Todo(descriptor);
+        } else if (words[0].equals("deadline")) {
+            String by = descriptor.split("/by ")[1];
+            descriptor = descriptor.split("/by ")[0];
+            t = new Deadline(descriptor, by);
+        } else if (words[0].equals("event")) {
+            String to = descriptor.split("/to ")[1];
+            String from = descriptor.split(" /")[1];
+            descriptor = descriptor.split("/")[0];
+            t = new Event(descriptor, from, to);
+        } else {
+            throw new IndexOutOfBoundsException();
```

Comment:
You should try to indicate in the method declaration line 32 the different types of exceptions thrown